exo-go/src/app_deployer/index.go

```diff
@@ -0,0 +1,39 @@
+package appDeployer
+
+import (
+	"path/filepath"
+
+	"github.com/Originate/exosphere/exo-go/src/logger"
+	"github.com/Originate/exosphere/exo-go/src/terraform_file_helpers"
+	"github.com/Originate/exosphere/exo-go/src/types"
+)
+
+// AppDeployer contains information needed to deploy the application
+type AppDeployer struct {
+	AppConfig      types.AppConfig
+	ServiceConfigs map[string]types.ServiceConfig
+	Logger         *logger.Logger
+	AppDir         string
+	HomeDir        string
+	TerraformDir   string
+}
+
+// NewAppDeployer is the constructor for the appDeployer class
+func NewAppDeployer(appConfig types.AppConfig, serviceConfigs map[string]types.ServiceConfig, appDir, homeDir string) *AppDeployer {
+	return &AppDeployer{
+		AppConfig:      appConfig,
+		ServiceConfigs: serviceConfigs,
+		AppDir:         appDir,
+		HomeDir:        homeDir,
+		TerraformDir:   getTerraformDir(appDir),
```

Comment:
I think its best to keep minimal state (less the think about). This can be computed very quickly so I don't see any reason we need to cache it. The function can be public if needed elsewhere.